### PR TITLE
Fix expiration and signing for attestation discounts

### PR DESCRIPTION
**Pull request by dneilroth**
**What changed? Why?**
This PR applies 2 fixes to the attestation discount signing process:
1: set the signature expiration time equal to the current time + the expiration in seconds value. currently the expiration time is equal to the expiration in seconds value
2: sign the message using a signing utility that will not automatically add the ` "\x19Ethereum Signed Message:\n000000"` header.

**Notes to reviewers**

**How has it been tested?**
locally on testnet

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/2</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/utils/proofs/sybil_resistance.ts [hunk 1]
```diff
@@ -140,7 +144,7 @@ export async function sybilResistantUsernameSigning(
     const hasPreviouslyRegistered = await hasRegisteredWithDiscount(linkedAddresses, chainId);
     // if any linked address registered previously return an error
     if (hasPreviouslyRegistered) {
-      throw new Error('You have already claimed a username with a different address (onchain).');
+      throw new Error('You have already claimed a discounted basename (onchain).');
```
Comment (dneilroth) [inline apps/web/src/utils/proofs/sybil_resistance.ts:147]: the connected address is included in `getLinkedAddresses` so its possible that the discounted name was already claimed with the connected address, not a different one

File: apps/web/src/utils/proofs/sybil_resistance.ts [hunk 2]
```diff
@@ -163,11 +166,12 @@ export async function sybilResistantUsernameSigning(
       };
     }
 
+    const expirationTimeUnix = Math.floor(Date.now() / 1000) + parseInt(EXPIRY);
```
Comment (dneilroth) [inline apps/web/src/utils/proofs/sybil_resistance.ts:169]: make sure that expiration is set to a time in the future, determined by the expiration in seconds env var

File: apps/web/src/utils/proofs/sybil_resistance.ts [hunk 3]
```diff
@@ -86,25 +86,29 @@ async function signMessageWithTrustedSigner(
...
 
   // sign the hashed message
-  const signature = await account.signMessage({ message: msgHash });
+  const { r, s, v } = await sign({
+    hash: msgHash,
+    privateKey: `0x${trustedSignerPKey}`,
+  });
+
+  // combine r, s, and v into a single signature
+  const signature = `${r.slice(2)}${s.slice(2)}${(v as bigint).toString(16)}`;
```
Comment (JFrankfurt) [inline apps/web/src/utils/proofs/sybil_resistance.ts:105]: I know you'd padded v to two chars in an earlier implementation--why the change?

**Review by JFrankfurt — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for JFrankfurt @ 2024-08-11 21:00:50 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by JFrankfurt — APPROVED** [review]

File: apps/web/src/utils/proofs/sybil_resistance.ts (hunk 3 above)
Comment (dneilroth) [reply apps/web/src/utils/proofs/sybil_resistance.ts:105]: did not seem necessary, but can go ahead and add back if that's safer 👍 my understanding is its typically either 27n or 28n.
